SG14/inplace_function.h

```diff
@@ -150,6 +150,76 @@ struct is_valid_inplace_dst : std::true_type
     );
 };
 
+// C++11 MSVC compatible implementation of std::is_invocable_r,
+// std version doesn't seem to respect return types and requires C++17.
+// See https://godbolt.org/z/T0bd3K for defect example.
```

Comment:
By "doesn't seem to respect return types ... defect example," IIUC, you mean this?
https://godbolt.org/z/YMT0PA

    static_assert( std::is_invocable_r_v<void, int()> );  // passes

That is, although `int` is not "convertible to" `void`, nonetheless `int()` is still "invocable with return type convertible to" `void`. This makes *some* intuitive sense, since you can always discard the result of a function call; and in fact this is what `std::function` does [[1]](https://quuxplusone.github.io/blog/2019/03/27/design-space-for-std-function/#can-i-implicitly-convert-f-int-t) [[2]](https://github.com/zenorogue/hyperrogue/commit/f6f96e1129df134a9921015dc6e59d45ce5a7e11). Some people probably want this behavior, so that you can write things like

    std::function<void(const char*)> f = std::puts;

But *I'm* an anecdotal example of someone who *wouldn't* want such a conversion to happen implicitly.

A priori, I'd recommend we follow `std::function`'s example and permit conversions such as

    stdext::inplace_function<void(const char*)> f =
        [](const char *) { return 0; };

even though it will make some people (such as myself) unhappy.

Reply:
Sigh, another day goes by and C++'s unreasonable compromises win once again.